src/analysis/normed_space/inner_product.lean

```diff
@@ -1374,6 +1382,56 @@ linear_map.mk_continuous
 
 end norm
 
+section bessels_inequality
+
+variables {ι: Type*} (x : E) {v : ι → E}
+
+lemma bessel_finite {s : finset ι} (hv : orthonormal 𝕜 v) : ∑ i in s, ∥ ⟪v i, x⟫ ∥^2 ≤ ∥x∥ ^2 :=
+begin
+  rw ←sub_nonneg,
+  suffices hbf: ∥ x -  ∑ i in s, ⟪v i, x⟫ • (v i) ∥^2 = ∥x∥^2 - ∑ i in s, ∥ ⟪v i, x⟫ ∥^2,
+  { rw ←hbf, simp only [norm_nonneg, pow_nonneg], },
+  rw [norm_sub_sq, sub_add], congr' 1,
+  simp only [inner_product_space.norm_sq_eq_inner, inner_sum],
+  simp only [sum_inner],
+  simp only [inner_smul_left, inner_smul_right, inner_conj_sym, ←mul_assoc],
+  have h₂ : ∑ i in s, ∑ i_1 in s, (inner (v i) x * inner x (v i_1)) * inner (v i_1) (v i)
+   = (∑ k in s, (inner (v k) x * inner x (v k)) : 𝕜 ),
+   { apply orthonormal.inner_left_right_finset, exact hv },
+  simp only [h₂], clear h₂,
+  simp only [add_monoid_hom.map_sum, finset.mul_sum, ←finset.sum_sub_distrib],
+  simp only [two_mul, add_sub_cancel'],
+  have h₃ : ∀ z : 𝕜, re (z * conj (z)) = ∥ z ∥ ^2,
+  { intro z, simp only [mul_conj, norm_sq_eq_def'], norm_cast, },
+  simp only [←h₃, inner_conj_sym],
```

Comment:
In this proof, if you move `h₂` and `h₃` upwards, I think you'll find that almost all of the last series of `simp only`s can be combined into a single `simp`.